app/models/claim/cleaner.rb

```diff
@@ -0,0 +1,50 @@
+class Claim
+  class Cleaner
+    attr_reader :claims
+
+    PII_ATTRIBUTES_TO_DELETE = [
+      :first_name,
+      :middle_name,
+      :surname,
+      :date_of_birth,
+      :address_line_1,
+      :address_line_2,
+      :address_line_3,
+      :address_line_4,
+      :postcode,
+      :payroll_gender,
+      :national_insurance_number,
+      :bank_sort_code,
+      :bank_account_number,
+      :building_society_roll_number,
+      :verify_response,
+    ]
+
+    def initialize(claims)
+      claims.each do |claim|
+        safe_to_clean(claim)
+      end
+      @claims = claims
+    end
+
+    def delete_pii
+      Claim.where(id: @claims.pluck(:id)).update_all(attribute_values_to_set)
```

Comment:
If we pass an `ActiveRecord_Relation` into the initializer, rather than an array, we can just do `@claims.update_all...`

Reply:
Actually, I'm wondering about moving this to the `Claim` model itself, so each `Claim` would have a `delete_pii!` method, then this class would be a very thin wrapper that loops through all the claims and calls `delete_pii!` on each.

Reply:
The only issue with that is that this is likely to impact a lot of claims simultaneously (ie every claim paid out in a cycle will all become eligible for cleaning at the same time), so doing an `update_all` saves running a DB query for each one.

I'm not totally convinced of the merits of doing that vs the method you suggest here though, which was my first thought.